daemon/cmd/daemon.go

```diff
@@ -1270,6 +1270,9 @@ func (d *Daemon) Close() {
 	}
 	identitymanager.RemoveAll()
 	d.cgroupManager.Close()
+
+	// Ensures all controllers are stopped!
+	d.controllers.RemoveAllAndWait()
```

Comment:
This is a separate fix altogether so it should be submitted separately in another PR.

Reply:
Not sure?? we introduce a new manager that would not properly stop when the daemon exits? 
Based on Jussi's comment above, I've gathered this could entail leaks for some of our tests?

Reply:
The reality is that this code is not present in main, so adding another controller on top is not going to change the fact that this gap already exists. We might as well submit a separate fix to get it merged and backported, while this PR is being reviewed.

Reply:
Tx Chris! makes sense